core/trino-main/src/main/java/io/trino/dispatcher/LocalDispatchQuery.java

```diff
@@ -270,6 +271,14 @@ public QueryInfo getFullQueryInfo()
                 .orElseGet(() -> stateMachine.updateQueryInfo(Optional.empty()));
     }
 
+    @Override
+    public Optional<Statement> getStatement()
+    {
+        return tryGetQueryExecution()
+                .map(QueryExecution::getStatement)
+                .orElseGet(() -> Optional.empty());
```

Comment:
I guess this is not required. `Optional#map` returns `Optional.empty if the element is not present.

Reply:
`.map(QueryExecution::getStatement)` would need to become `flatMap` (as `QueryExecution::getStatement` returns an optional)